Clean up xpu ut to make CI happy

guangyey opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #128383

# Motivation
Before #127611 merged, the xpu-specific UT `test/test_xpu.py` was skipped temporarily. This PR aims to fix the UT bug introduced by #127741.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

guangyey commented on test/test_xpu.py:
```diff
@@ -312,6 +312,11 @@ def test_serialization_array_with_empty(self):
+    # These operators are not implemented on XPU backend and we can NOT fall back
+    # them to CPU. So we have to skip them at this moment.
+    # TODO: remove these operators from skip list when they are implemented on XPU backend.
+    skip_list = ["gru_cell"]
```
`_thnn_fused_gru_cell` has not a CPU implementation.

etaf replied:
_thnn_fused_gru_cell has not XPU nor CPU implementation.

guangyey commented:
UT pass
```
Running test_xpu 1/1 ... [2024-06-11 08:43:42.024503]
Executing ['/opt/conda/envs/py_3.8/bin/python', '-bb', 'test_xpu.py', '-m', 'serial', '--shard-id=1', '--num-shards=1', '-v', '-vv', '-rfEX', '-p', 'no:xdist', '--use-pytest', '-x', '--reruns=2', '--import-slow-tests', '--import-disabled-tests'] ... [2024-06-11 08:43:42.024838]

test_xpu 1/1 was successful, full logs can be found in artifacts with path test/test-reports/test_xpu_1.1_8864c4c5d0f7c42a_.log
```

guangyey committed:
fix xpu ut bug

[ghstack-poisoned]

guangyey committed:
Update on "fix xpu ut bug"




[ghstack-poisoned]

guangyey committed:
Update on "fix xpu ut bug"




[ghstack-poisoned]

guangyey committed:
Update on "Clean up xpu ut to make CI happy"


# Motivation
Before #127611 merged, the xpu-specific UT `test/test_xpu.py` was skipped temporarily. This PR aims to fix the UT bug introduced by #127741.

[ghstack-poisoned]

EikanWang commented:
@pytorchbot merge

etaf commented:
@pytorchmergebot cherry-pick --onto release/2.4 -c critical
